subprojects/docs/src/docs/userguide/upgrading_version_5.adoc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[upgrading_version_5]]
+= Upgrading your build from Gradle 5.0
+
+This chapter provides the information you need to migrate your Gradle 5.0 builds to Gradle 5.1.
+
+We recommend the following steps for all users:
+
+. Try running `gradle help --scan` and view the https://gradle.com/enterprise/releases/2018.4/#identify-usages-of-deprecated-gradle-functionality[deprecations view] of the generated build scan.
```

Comment:
I'm conflicted here because I want to promote build scans deprecations, but we only remove deprecated stuff on major versions and so this might not be the most useful thing for users to do until they upgrade to Gradle 6.0. 

I do think some general guidance for upgrading is good here, but I cannot think of anything other than "upgrade your plugins, here's how: (bulleted list)"

Reply:
I think it's not a bad idea. Even if it's just warnings, it's creating a habit of checking.

Reply:
@eriwen Do you want to discuss this further or should I merge it since I resolved your other suggestions?

Reply:
Better to merge @jlstrater. Sorry for the delayed response.